packages/frontend/editor-ui/src/components/canvas/experimental/components/ExperimentalEmbeddedNodeDetails.vue

```diff
@@ -0,0 +1,173 @@
+const nodeTypesStore = useNodeTypesStore();
+const workflowsStore = useWorkflowsStore();
+const node = computed(() => workflowsStore.getNodeById(nodeId) ?? null);
+const nodeType = computed(() => {
+	if (node.value) {
+		return nodeTypesStore.getNodeType(node.value.type, node.value.typeVersion);
+	}
+	return null;
+});
+const vf = useVueFlow(workflowsStore.workflowId);
+
+const isMoving = ref(false);
+
+const moveStartListener = vf.onMoveStart(() => {
+	isMoving.value = true;
+});
+
+const moveEndListener = vf.onMoveEnd(() => {
+	isMoving.value = false;
+});
+
+onBeforeUnmount(() => {
+	moveStartListener.off();
+	moveEndListener.off();
+});
+
+const isVisible = computed(() =>
+	vf.isNodeIntersecting(
+		{ id: nodeId },
+		{
+			x: -vf.viewport.value.x / vf.viewport.value.zoom,
+			y: -vf.viewport.value.y / vf.viewport.value.zoom,
+			width: vf.viewportRef.value?.offsetWidth ?? 0,
+			height: vf.viewportRef.value?.offsetHeight ?? 0,
+		},
+	),
+);
+const isOnceVisible = ref(isVisible.value);
+
+watchOnce(isVisible, (visible) => {
+	isOnceVisible.value = isOnceVisible.value || visible;
+});
+
+function handleToggleExpand() {
+	experimentalNdvStore.setNodeExpanded(nodeId);
+}
+</script>
+
+<template>
+	<div
+		ref="container"
+		:class="[$style.component, isExpanded ? $style.expanded : $style.collapsed]"
+		:style="{ '--zoom': `${1 / experimentalNdvStore.maxCanvasZoom}` }"
+	>
+		<template v-if="isOnceVisible">
+			<ExperimentalCanvasNodeSettings
+				v-if="isExpanded"
+				:node-id="nodeId"
+				:class="$style.settingsView"
+				:no-wheel="
+					!isMoving /* to not interrupt panning while allowing scroll of the settings pane, allow wheel event while panning */
+				"
+			>
+				<template #actions>
+					<N8nIconButton
+						icon="compress"
+						type="secondary"
+						text
+						size="mini"
+						icon-size="large"
+						aria-label="Toggle expand"
+						@click="handleToggleExpand"
+					/>
+				</template>
+			</ExperimentalCanvasNodeSettings>
+			<div v-else role="button " :class="$style.collapsedContent" @click="handleToggleExpand">
+				<NodeTitle
+					v-if="node"
+					class="node-name"
+					:model-value="node.name"
+					:node-type="nodeType"
+					read-only
+				/>
+				<N8nIcon icon="expand" size="large" />
+			</div>
+		</template>
+	</div>
+</template>
+
+<style lang="scss" module>
+:root .component {
+	position: relative;
+	align-items: flex-start;
+	justify-content: stretch;
+	overflow: visible;
+	border-width: 0 !important;
+	outline: none;
+	box-shadow: none !important;
+	background-color: transparent;
+	width: calc(var(--canvas-node--width) * 1.5);
+
+	&.expanded {
+		cursor: default;
+	}
+
+	&.collapsed {
+		height: 50px;
+		margin-block: calc(var(--canvas-node--width) * 0.25);
+	}
+}
+
+:root :global(.vue-flow__node):has(.component) {
+	z-index: 10;
+
+	:global(.selected) & {
+		z-index: 11;
+	}
+}
+
+:root .collapsedContent,
+:root .settingsView {
+	border-radius: 4px;
```

Comment:
I appreciate this is an experimental feature but we should be using our variables over magic numbers as much as possible in case we move it in production one day.
```suggestion
	border-radius: var(--border-radius-base);
```